src/pages/index.js

```diff
@@ -62,17 +115,104 @@ cardAddBtn.addEventListener("click", () => {
   newCardPopup.open();
 });
 
+pfp.addEventListener("click", () => {
+  EditPfp.open();
+});
+
 //Handler
+function handleDeleteCard(card) {
+  confirmPopup.open();
+  confirmPopup.handleConfirm(() => {
+    confirmPopup.showLoading(true);
+    api
+      .deleteCard(card.getCardId())
+      .then(() => {
+        card.handleDeleteCard();
+        confirmPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmPopup.showLoading(false);
+      });
+  });
+}
+
 function editProfileSubmit(userData) {
+  profilePopup.showLoading(true);
+  api
+    .editUserData({
+      name: userData.name,
+      description: userData.description,
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.showLoading(false);
+    });
   userInfo.setUserInfo(userData);
   profilePopup.close();
 }
 
+function editPfpSubmit(userData) {
+  EditPfp.showLoading(true);
+  api
+    .editPfp(userData)
+    .then((res) => {
+      userInfo.setUserPfp({ avatar: res.avatar });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      EditPfp.showLoading(false);
+    });
+  formValidators["pfp-editor"].disableButton();
```

Comment:
You need to `disable` the button only after **resetting** the form when you have submitted a card/avatar in block `then` otherwise you'll not be able to click it if there is a server error